br/pkg/backup/client.go

```diff
@@ -408,9 +408,18 @@ func (bc *Client) BuildBackupRangeAndSchema(
 	tableFilter filter.Filter,
 	backupTS uint64,
 	isFullBackup bool,
-) ([]rtree.Range, *Schemas, []*backuppb.PlacementPolicy, error) {
+) ([]rtree.Range, *SchemasV2, []*backuppb.PlacementPolicy, error) {
 	if bc.checkpointMeta == nil {
-		return BuildBackupRangeAndSchema(storage, tableFilter, backupTS, isFullBackup, true)
+		ranges, schemas, policies, err := BuildBackupRangeAndSchema(storage, tableFilter, backupTS, isFullBackup, true)
+		if err != nil {
+			return nil, nil, nil, errors.Trace(err)
+		}
+		// Add keyspace prefix to BackupRequest
+		for i := range ranges {
+			start, end := ranges[i].StartKey, ranges[i].EndKey
+			ranges[i].StartKey, ranges[i].EndKey = storage.GetCodec().EncodeRange(start, end)
+		}
+		return ranges, schemas, policies, err
```

Comment:
Note: change the ranges with keyspace prefix only when get the ranges from kv storage instead of checkpoint data.